src/ext/ddtrace.c

```diff
@@ -405,6 +416,100 @@ static PHP_RSHUTDOWN_FUNCTION(ddtrace) {
+    php_info_print_table_row(2, "DATADOG TRACER CONFIGURATION", ZSTR_VAL(buf.s));
+#else
+    php_info_print_table_row(2, "DATADOG TRACER CONFIGURATION", buf.c);
+#endif
+    smart_str_free(&buf);
+}
+
+static void _dd_info_diagnostics_row(const char *key, const char *value TSRMLS_DC) {
+    if (sapi_module.phpinfo_as_text) {
+        php_info_print_table_row(2, key, value);
+        return;
+    }
+    datadog_info_print("<tr><td class='e'>" TSRMLS_CC);
+    datadog_info_print(key TSRMLS_CC);
+    datadog_info_print("</td><td class='v' style='background-color:#f0e881;'>" TSRMLS_CC);
+    datadog_info_print(value TSRMLS_CC);
+    datadog_info_print("</td></tr>" TSRMLS_CC);
+}
+
+static void _dd_info_diagnostics_table(TSRMLS_D) {
+    php_info_print_table_start();
+    php_info_print_table_colspan_header(2, "Diagnostics");
+
+    HashTable *ht;
+    ALLOC_HASHTABLE(ht);
+    zend_hash_init(ht, 8, NULL, ZVAL_PTR_DTOR, 0);
+
+    ddtrace_startup_diagnostics(ht);
+
+#if PHP_VERSION_ID >= 70000
+    zend_string *key;
+    zval *val;
+    ZEND_HASH_FOREACH_STR_KEY_VAL_IND(ht, key, val) {
+        switch (Z_TYPE_P(val)) {
+            case IS_STRING:
+                _dd_info_diagnostics_row(ZSTR_VAL(key), Z_STRVAL_P(val) TSRMLS_CC);
+                break;
+            case IS_NULL:
+                _dd_info_diagnostics_row(ZSTR_VAL(key), "NULL" TSRMLS_CC);
+                break;
+            case IS_TRUE:
+            case IS_FALSE:
+                _dd_info_diagnostics_row(ZSTR_VAL(key), Z_TYPE_P(val) == IS_TRUE ? "true" : "false" TSRMLS_CC);
+                break;
+            default:
+                _dd_info_diagnostics_row(ZSTR_VAL(key), "{unknown type}" TSRMLS_CC);
+                break;
+        }
+    }
+    ZEND_HASH_FOREACH_END();
+#else
+    int key_type;
+    zval **val;
+    HashPosition pos;
+    char *key;
+    uint key_len;
+    ulong num_key;
+    zend_hash_internal_pointer_reset_ex(ht, &pos);
+    while (zend_hash_get_current_data_ex(ht, (void **)&val, &pos) == SUCCESS) {
+        key_type = zend_hash_get_current_key_ex(ht, &key, &key_len, &num_key, 0, &pos);
+        if (key_type == HASH_KEY_IS_STRING) {
+            switch (Z_TYPE_PP(val)) {
+                case IS_STRING:
+                    _dd_info_diagnostics_row(key, Z_STRVAL_PP(val) TSRMLS_CC);
+                    break;
+                case IS_NULL:
+                    _dd_info_diagnostics_row(key, "NULL" TSRMLS_CC);
+                    break;
+                case IS_BOOL:
+                    _dd_info_diagnostics_row(key, Z_BVAL_PP(val) ? "true" : "false" TSRMLS_CC);
+                    break;
+                default:
+                    _dd_info_diagnostics_row(key, "{unknown type}" TSRMLS_CC);
+                    break;
+            }
+        }
+        zend_hash_move_forward_ex(ht, &pos);
+    }
+#endif
+
+    if (zend_hash_num_elements(ht) == 0) {
+        php_info_print_table_row(2, "All diagnostic checks", "passed");
+    }
```

Comment:
Would it be better to always have this row and have passed/did not pass?

Reply:
Good point. I updated this. 👍